install/uninstall.ps1

```diff
@@ -0,0 +1,95 @@
+# Enable a default -Verbose flag for debug output.
+[CmdletBinding()]
+
+# Halt script execution at the first failed command.
+$script:ErrorActionPreference='Stop'
+
+$deleted = $false
+
+function Uninstall {
+    $uninstallDirectory = Get-UninstallDirectory
+    Remove-InstallationFolder $uninstallDirectory
+    Remove-FromPath $uninstallDirectory
+
+    if ($deleted) {
+        Write-Output "Uninstalled AzureAuth!"
+    } else {
+        Write-Output "There were no AzureAuth installations to be deleted."
+    }
+}
+
+function Get-UninstallDirectory {
+    $directory = if (![string]::IsNullOrEmpty($Env:AZUREAUTH_INSTALL_DIRECTORY)) {
+        Write-Error "Uninstalling from custom location is not supported"
+    } 
+    return ([System.IO.Path]::Combine($Env:LOCALAPPDATA, "Programs", "AzureAuth"))
+}
+
+function Remove-InstallationFolder {
+    param ([string]$directory)
+
+    if (Test-Path -Path $directory) {
+        Write-Verbose "Removing installations at '${directory}'"
+        Remove-Item -Force -Recurse $directory
+        $script:deleted = $true
+    } else {
+        Write-Verbose "There were no installations found at '${directory}'"
+    }
+}
+
+function Remove-FromPath {
+    param ([string]$uninstallDirectory)
+
+    $registryPath = 'Registry::HKEY_CURRENT_USER\Environment'
+    $currentPath = (Get-ItemProperty -Path $registryPath -Name PATH -ErrorAction SilentlyContinue).Path
+
+    # We try to find any other AzureAuth installations that are not in default location.
+    # Installations in custom locations that are not listed in PATH cannot be found.
+    $azureauthsInPath = (Get-Command -Name azureauth -ErrorAction SilentlyContinue -CommandType Application -All).Source
+    $otherDirectories = [System.Collections.ArrayList]@()
+    ForEach($az in $azureauthsInPath) {
+        if (!$az.Contains($uninstallDirectory)) {
+            $additionalDirectory = (Get-Item $az).Directory.FullName
+            Write-Verbose "Additional installation found in ${additionalDirectory}"
+            $_ = $otherDirectories.Add($additionalDirectory)
+            Remove-InstallationFolder ($additionalDirectory)

```

Comment:
Should we re-design this so `Remove-FromPath` only updates the path and does not perform any folder deletion? We can first gather these custom installations and call `Remove-InstalllationFolder`?

Reply:
Will do, thank you!

Reply:
Remove-FromPath doesn't do folder deletions anymore.